libcontainerd/types/types_linux.go

```diff
@@ -27,7 +27,7 @@ func InterfaceToStats(read time.Time, v interface{}) *Stats {
 }
 
 // Resources defines updatable container resource values. TODO: it must match containerd upcoming API
-type Resources specs.LinuxResources
+type Resources = specs.LinuxResources
```

Comment:
Was this needed for this PR? I feel like this should be separate (and possibly when doing so, mark `Resources` deprecated in favour of `specs.LinuxResources`

Reply:
It avoids having to explicitly cast `types.Resources` values to `specs.LinuxResources`. And the type serves a purpose: it allows the libcontainerd `Task` interface to be have a single definition which is applicable for both Windows and Linux platforms. https://github.com/moby/moby/blob/1aae94074e63736a4b419f08cddc1baaaa717453/libcontainerd/types/types_windows.go#L26-L27
https://github.com/moby/moby/blob/1aae94074e63736a4b419f08cddc1baaaa717453/libcontainerd/types/types.go#L96

I am sure it would have been written as a type alias originally, if it existed at the time. (Proper type aliases were added in Go 1.9.) https://github.com/moby/moby/blob/1aae94074e63736a4b419f08cddc1baaaa717453/libcontainerd/remote/client_linux.go#L24-L26

I can split this out into its own commit if you feel strongly about it, but I don't think it's a significant enough change to warrant a separate PR. While technically a breaking change in the strictest sense, even Go's own apidiff tooling [does not consider merging types to be worth flagging.](https://pkg.go.dev/golang.org/x/exp/apidiff#section-readme:~:text=type%20potentially%20incompatible.-,Type%20Switches,-A%20package%20change)